llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -995,6 +995,44 @@ Value *InstCombinerImpl::SimplifyDemandedUseBits(Value *V, APInt DemandedMask,
                 I, 1, (DemandedMask & ~LHSKnown.Zero).zextOrTrunc(MaskWidth)))
           return I;
 
+        // Combine:
+        // (ptrmask (getelementptr i8, ptr p, imm i), imm mask)
+        //   -> (ptrmask (getelementptr i8, ptr p, imm (i & mask)), imm mask)
+        // where only the low bits known to be zero in the pointer are changed
+        Value *InnerPtr;
+        uint64_t GEPIndex;
+        uint64_t PtrMaskImmediate;
+        if (match(I, m_Intrinsic<Intrinsic::ptrmask>(
+                         m_PtrAdd(m_Value(InnerPtr), m_ConstantInt(GEPIndex)),
+                         m_ConstantInt(PtrMaskImmediate)))) {
+
+          const unsigned trailingZeros =
+              computeKnownBits(InnerPtr, Depth + 1, I).countMinTrailingZeros();
+
+          if (trailingZeros < 64) {
```

Comment:
Or `Q.DL.getIndexTypeSizeInBits` actually.

Reply:
"Resolve" does not apply a comment, merely deletes the contents of the text box and then hides it.

The 64 magic number here was avoiding shifting a uin64_t by 64 (because C++), replaced with an explicit zero test. Added a test, the zero case is indeed handled correctly elsewhere in instcombine.